### PR TITLE
Apply request extensions on forged requests (ref Kinto/kinto#438)

### PR DESCRIPTION
**Pull request by leplatrem**

### CODE REVIEW

**Comment by leplatrem** [comment]
pair=@almet+@Natim

File: cliquet/tests/test_utils.py [hunk 1]
```diff
@@ -118,6 +119,26 @@ def test_current_service_returns_none_for_unexisting_patterns(self):
         self.assertEqual(current_service(request), None)
 
 
+def build_real_request(wsgi_environ):
+    from pyramid import request as pyramid_request
```
Comment (Natim) [inline cliquet/tests/test_utils.py]: Don't you want to put this on the top of the file?

File: cliquet/tests/test_utils.py [hunk 2]
```diff
@@ -118,6 +119,26 @@ def test_current_service_returns_none_for_unexisting_patterns(self):
...
+    request.registry = config.registry
+    return request
+
+
+class BuildRequestTest(unittest.TestCase):
+
+    def test_built_request_has_cliquet_custom_methods(self):
+        original = build_real_request({'PATH_INFO': '/foo'})
+        request = build_request(original, {"path": "bar"})
+        self.assertIsNone(request.current_service)
```
Comment (almet) [inline cliquet/tests/test_utils.py]: shouldn't this be ``assertIsNotNone`` to comply with the name of the test?

Comment (almet) [inline cliquet/tests/test_utils.py]: Ah, nevermind. I found it weird to have it assert that it returns None. Can be replaced with `assertTrue(hasattr(request, 'current_server'))` for ease of read.

File: cliquet/utils.py [hunk 3]
```diff
@@ -262,6 +262,10 @@ def build_request(original, dict_obj):
                             POST=payload,
                             method=method)
 
+    request.registry = original.registry
+
```
Comment (almet) [inline cliquet/utils.py]: extra space here.

**Comment by almet** [comment]
LGTM with nits.

**Comment by Natim** [comment]
r+

**Pull request merged**